src/scan.l

```diff
@@ -682,7 +683,19 @@ F	RET_CONST(zeek::val_mgr->False()->Ref())
 
 "0x"{HEX}+	RET_CONST(zeek::val_mgr->Count(static_cast<zeek_uint_t>(strtoull(yytext, 0, 16))).release())
 
-({H}".")+{HTLD}		RET_CONST(zeek::detail::dns_mgr->LookupHost(yytext).release())
+({H}".")+{HTLD} {
+	zeek::TableValPtr result = zeek::detail::dns_mgr->empty_addr_set();
+
+	if ( ! zeek::detail::parse_only ) {
+		const char *allow = getenv("ZEEK_DNS_ALLOW_SCRIPT_RESOLVE");
+		if ( ! zeek::fake_dns() && (! allow || std::string(allow) == "0") )
```

Comment:
Is the `zeek::fake_dns` case in here to allow tests to work without setting the environment variable, or is there a different reason?

Reply:
Mostly that and the fact that ZEEK_FAKE_DNS won't do any real DNS resolutions, so seemed okay to allow.

But it's a good point, I can also see ditching `zeek::fake_dns()` here and adapt the tests - seems more consistent - is that what you had in mind?

Reply:
Yup, that was where my question was going,mostly for consistency.